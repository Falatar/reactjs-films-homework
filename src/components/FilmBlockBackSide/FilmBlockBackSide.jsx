import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import style from './FilmBlockBackSide.scss';
import FilmDescription from '../FilmDescription';

function FilmBlockBackSide({
  name, tagList, rating, callModal, switchMode, poster, overview,
}) {
  return (
    <div
      className={classNames(style.film__block, style.dark)}
      style={{ backgroundImage: `url(${poster})` }}
    >
      <button type="button" className={style.close} onClick={switchMode}>âœ•</button>
      <FilmDescription
        name={name}
        tagList={tagList}
        rating={rating}
        mode={false}
      />
      <p className={style.overview}>{overview}</p>
      <button type="button" className={style.details_play} onClick={callModal}>Watch Now</button>
    </div>
  );
}

FilmBlockBackSide.defaultProps = {
  name: 'Can\'t find property "name"',
  poster: '',
  rating: 0,
  tagList: 'Can\'t find property "tagList"',
  overview: 'Can\'t find property "overview"',
  callModal: () => {},
  switchMode: () => {},
};

FilmBlockBackSide.propTypes = {
  name: PropTypes.string,
  poster: PropTypes.string,
  rating: PropTypes.number,
  tagList: PropTypes.string,
  overview: PropTypes.string,
  callModal: PropTypes.func,
  switchMode: PropTypes.func,
};

export default FilmBlockBackSide;
