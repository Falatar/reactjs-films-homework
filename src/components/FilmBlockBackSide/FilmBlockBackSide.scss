.dark::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0,0,0,0.664);
    z-index: 1;
}  

.film__block {

    height: 35vh;
    width: 46vh;
    background-color: white;
    margin: 1%;
    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    background-color: white;
    background-size: cover;

    .close {
        width: 15%;
        height: 10%;
        background: transparent;
        border: 0;
        color: white;
        font-size: 1.5em;
        font-weight: bold;
        outline: none;
        z-index: 2;
        margin-bottom: 5%;;
    }

    .close:active {
        color: black;
    }

    .overview {
        height: 50%;
        color: white;
        z-index: 2;
        margin: 3% 5%;
        font-size: 1.2em;
        text-overflow: ellipsis;
        resize: none;
        overflow-y: auto;
        scrollbar-color: deepskyblue #201F27;
    }

    .details_play {
        width: 40%;
        height: 15%;
        border: 0;
        border-radius: 10px;
        z-index: 2;
        color: white;
        background-color: deepskyblue;
        font-size: 1.3em;
        margin: 5% 0 5% 5%;
        outline: none;
    }

    .details_play:active {
        background-image: -webkit-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
        background-image: -moz-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
        background-image: -o-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
        background-image: -ms-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
        background-image: linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
    }
}

::-webkit-scrollbar-button {
    background-image:url('');
    background-repeat:no-repeat;
    width:6px;
    height:0px
    }
    
::-webkit-scrollbar-track {
    background-color:#201F27;
    box-shadow:0px 0px 3px #000 inset;
}
    
::-webkit-scrollbar-thumb {
    -webkit-border-radius: 5px;
    border-radius: 5px;
    background-color: deepskyblue;
    box-shadow:0px 1px 1px #201F27 inset;
    background-image:url('https://yraaa.ru/_pu/24/59610063.png');
    background-position:center;
    background-repeat:no-repeat;
}
    
::-webkit-resizer{
    background-image:url('');
    background-repeat:no-repeat;
    width:7px;
    height:0px
}
    
::-webkit-scrollbar{
    width: 11px;
}