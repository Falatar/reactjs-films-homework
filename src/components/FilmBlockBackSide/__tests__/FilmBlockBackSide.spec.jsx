import React from 'react';
import renderer from 'react-test-renderer';
import FilmBlockBackSide from '../FilmBlockBackSide';

jest.mock('../../FilmDescription', () => 'FilmDescription');

describe('FilmBlockBackSide', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <FilmBlockBackSide
          poster=""
          name="nemo"
          tagList="empty"
          rating={0.0}
          overview="some text"
          callModal={() => {}}
          switchMode={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
