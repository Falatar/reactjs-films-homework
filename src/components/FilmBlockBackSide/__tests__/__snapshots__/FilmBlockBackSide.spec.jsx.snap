// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmBlockBackSide renders correctly 1`] = `
<div
  className="film__block dark"
  style={
    Object {
      "backgroundImage": "url()",
    }
  }
>
  <button
    className="close"
    onClick={[Function]}
    type="button"
  >
    âœ•
  </button>
  <FilmDescription
    mode={false}
    name="nemo"
    rating={0}
    tagList="empty"
  />
  <p
    className="overview"
  >
    some text
  </p>
  <button
    className="details_play"
    onClick={[Function]}
    type="button"
  >
    Watch Now
  </button>
</div>
`;
